sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/CoverageReportImportSensor.java

```diff
@@ -101,25 +111,32 @@ void analyze(SensorContext context, Coverage coverage) {
 
       if (inputFile.type().equals(Type.TEST)) {
         fileCountStatistics.test++;
+        LOG.debug("Will skip '{}' as it is a test file.", filePath);
         // Do not log for test files to avoid pointless noise
         continue;
       }
 
       if (!coverageConf.languageKey().equals(inputFile.language())) {
+        LOG.debug("Will skip '{}' as conf lang '{}' does not equal file lang '{}'.",
+          filePath, coverageConf.languageKey(), inputFile.language());
         fileCountStatistics.otherLanguageExcluded++;
         continue;
       }
 
+      LOG.debug("Will check main file coverage for '{}'.", filePath);
       fileCountStatistics.main++;
       boolean fileHasCoverage = false;
 
       NewCoverage newCoverage = context.newCoverage().onFile(inputFile);
       for (Map.Entry<Integer, Integer> entry : coverage.hits(filePath).entrySet()) {
+        LOG.debug("Found entry with key '{}' and value '{}'.", entry.getKey(), entry.getValue());
         fileHasCoverage = true;
         newCoverage.lineHits(entry.getKey(), entry.getValue());
       }
       newCoverage.save();
 
+      LOG.debug("fileHasCoverage for '{}' is '{}'.", filePath, fileHasCoverage);
```

Comment:
I suggest to remove this line as duplicate. Both `true` and `false` scenarios have more detailed log messages just below. The term `fileHasCoverage` is hard to reach for average message reader because it's internal variable name (not configuration parameter).

Reply:
good catch!